teeio-validator/library/pcie_ide_test_lib/test_case/test_case_keyprog_2.c

```diff
@@ -59,58 +56,67 @@ bool test_ide_km_key_prog_case2(const void *pci_doe_context,
+    left_assertion_cnt--;
+    ret = res && ret;
 
     // Assertion.3
     res = response.status == PCI_IDE_KM_KP_ACK_STATUS_INCORRECT_LENGTH;
-    TEEIO_PRINT(("         %s.3: %s(0x%x) %s\n", case_msg, mKeyProgAssersion[3], status, res ? "Pass" : "Fail"));
-    if(!res) {
-        return false;
-    }
+    assertion_result = res ? IDE_COMMON_TEST_CASE_ASSERTION_RESULT_SUCCESS : IDE_COMMON_TEST_CASE_ASSERTION_RESULT_FAILED;
+    teeio_record_assertion_result(case_class, case_id, 3, IDE_COMMON_TEST_CASE_ASSERTION_TYPE_TEST, assertion_result, "IdeKmMessage.Status = 0x%02x", response.status);
+    left_assertion_cnt--;
+    ret = res && ret;
 
     // Assertion.4
     res = (response.port_index == request.port_index);
-    TEEIO_PRINT(("         %s.3: %s %s\n", case_msg, mKeyProgAssersion[4], res ? "Pass" : "Fail"));
-    if(!res) {
-        return false;
-    }
+    assertion_result = res ? IDE_COMMON_TEST_CASE_ASSERTION_RESULT_SUCCESS : IDE_COMMON_TEST_CASE_ASSERTION_RESULT_FAILED;
+    teeio_record_assertion_result(case_class, case_id, 4, IDE_COMMON_TEST_CASE_ASSERTION_TYPE_TEST, assertion_result, "IdeKmMessage.PortIndex = 0x%x", response.port_index);
+    left_assertion_cnt--;
+    ret = res && ret;
 
     // Assertion.5
     res = (response.stream_id == request.stream_id);
-    TEEIO_PRINT(("         %s.2: %s %s\n", case_msg, mKeyProgAssersion[5], res ? "Pass" : "Fail"));
-    if(!res) {
-        return false;
-    }
+    assertion_result = res ? IDE_COMMON_TEST_CASE_ASSERTION_RESULT_SUCCESS : IDE_COMMON_TEST_CASE_ASSERTION_RESULT_FAILED;
+    teeio_record_assertion_result(case_class, case_id, 5, IDE_COMMON_TEST_CASE_ASSERTION_TYPE_TEST, assertion_result, "IdeKmMessage.StreamID = 0x%x", response.stream_id);
+    left_assertion_cnt--;
+    ret = res && ret;
 
     // Assertion.6
     res = (response.key_sub_stream == request.key_sub_stream);
-    TEEIO_PRINT(("         %s.2: %s %s\n", case_msg, mKeyProgAssersion[6], res ? "Pass" : "Fail"));
-    if(!res) {
-        return false;
+    assertion_result = res ? IDE_COMMON_TEST_CASE_ASSERTION_RESULT_SUCCESS : IDE_COMMON_TEST_CASE_ASSERTION_RESULT_FAILED;
+    teeio_record_assertion_result(case_class, case_id, 6, IDE_COMMON_TEST_CASE_ASSERTION_TYPE_TEST, assertion_result, 
+                                  "IdeKmMessage.KeySet = 0x%02x && IdeKmMessage.RxTx = 0x%02x && IdeKmMessage.SubStream = 0x%02x",
+                                  response.key_sub_stream & PCI_IDE_KM_KEY_SET_MASK,
+                                  response.key_sub_stream & PCI_IDE_KM_KEY_DIRECTION_MASK,
+                                  response.key_sub_stream & PCI_IDE_KM_KEY_SUB_STREAM_MASK);
+    left_assertion_cnt--;
+    ret = res && ret;
+
+PcieIdeKeyProg2Done:
+    for(int i = 0; i < left_assertion_cnt; i++) {
+        teeio_record_assertion_result(case_class, case_id, i + 1, IDE_COMMON_TEST_CASE_ASSERTION_TYPE_TEST, IDE_COMMON_TEST_CASE_ASSERTION_RESULT_SKIPPED, "");
     }
```

Comment:
This style is not 100% correct, because we can skip some assertion in the middle.